From 5db9f45c041511ae41fac1989f05cd30e3a7cbde Mon Sep 17 00:00:00 2001
From: Matt Thompson <mattt@defunct.ca>
Date: Fri, 4 Dec 2015 15:31:17 +0000
Subject: [PATCH] Add run_once to 'copy keys to the ansible server'

I have seen a number of failures on this task due to mismatch of
checksum of source file and destination.  I suspect this is due to a
race condition caused by several hosts simultaneously copying the same
file to single location on the deployment server.

This change simply updates the 'copy keys to the ansible server' task
by adding 'run_once', which limits the task to being run on a single
MON host.

Closes issue #410
---
 tasks/ceph_keys.yml | 1 +
  1 file changed, 1 insertion(+)

diff --git a/tasks/ceph_keys.yml b/tasks/ceph_keys.yml
index a28aca4..544478c 100644
--- a/tasks/ceph_keys.yml
+++ b/tasks/ceph_keys.yml
@@ -38,6 +38,7 @@
     src={{ item }}
     dest={{ fetch_directory }}/{{ fsid }}/{{ item }}
     flat=yes
+  run_once: true
   with_items:
     - "{{ ceph_keys.stdout_lines }}"
     - /var/lib/ceph/bootstrap-osd/ceph.keyring
